cs169/BJC-Teacher-Tracker-App
# Teacher homeschool feature

### [Pivotal Tracker Link][tracker]


[tracker]: [https://www.pivotaltracker.com/story/show/187479544](https://www.pivotaltracker.com/story/show/187479544), [https://www.pivotaltracker.com/story/show/187460461](https://www.pivotaltracker.com/story/show/187460461), [https://www.pivotaltracker.com/story/show/187460457](https://www.pivotaltracker.com/story/show/187460457), [https://www.pivotaltracker.com/story/show/187460456](https://www.pivotaltracker.com/story/show/187460456)

## What this PR does:


A summary of relevant changes:

- Adds a status option "I am teaching homeschool with the BJC curriculum" to the status dropdown
- Iff the homeschool status option is selected, the teacher has the option to upload supporting files
- The teacher can upload additional supporting files by editing their profile
- The admin can both add and delete any teacher's supporting files

A future PR might add functionality to allow teachers to delete files that they uploaded by. This might be done by, for example, "marking" files to be purged when a teacher updates their profile from the edit teacher page.

This pull request fixes|implements (pick one...) ______.

### Include screenshots, videos, etc.

<img width="1436" alt="Screen Shot 2024-04-23 at 1 41 59 PM" src="https://github.com/beautyjoy/BJC-Teacher-Tracker/assets/65872167/46fbb494-ba85-41c6-a010-1ad2f10ca0f9">

<img width="1136" alt="Screen Shot 2024-04-23 at 1 42 22 PM" src="https://github.com/beautyjoy/BJC-Teacher-Tracker/assets/65872167/0ddc49e2-d844-43d8-b979-569c4d716d0f">


#### Who authored this PR?


@ArushC 

### How should this PR be tested?

Look at the Cucumber tests to verify that everything works as intended.

* Is there a deploy we can view?

The Heroku deployment is currently buggy. I am working to resolve issues, and I will update this description when it gets figured out.

* What do the specs/features test?

* Are there edge cases to watch out for?

#### Are there any complications to deploying this?

Nope



### Checklist:
